dev/MRTCore/mrt/MrtCore.sln

```diff
@@ -21,6 +21,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.ApplicationModel.
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MrtCoreUnpackagedTests", "Microsoft.ApplicationModel.Resources\UnpackagedTests\MrtCoreUnpackagedTests.csproj", "{3C618444-4B80-492E-8972-BFAEF8700A52}"
 EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Microsoft.WindowsAppSDK.Insights", "..\..\WindowsAppSDK_Insights\WindowsAppSDKInsightsLib\WindowsAppSDKInsightsLib.vcxproj", "{4243B8DA-BDB5-4D65-B1D5-EEED375FAD13}"
+EndProject
```

Comment:
why?

Reply:
The Insights lib and header are needed to log telemetry.

Reply:
Out of curiosity, by how much does the binary size increase due to this change?

(just give me the new number, I know how large it is before your change:))

Reply:
the lib contains minimal code. is it possible to make it header only?

Reply:
For Axel's question,
The new x64 debug mrm.dll is 898KB in size.
The old x64 debug mrm.dll is 840KB in size.

For Hui's question, redirecting to @ZenBird.

Reply:
If we make it header only, the code is included at all call-sites where the header is included. In the interest of memory/disk footprint, this code is distributed as a lib.

Reply:
Have you confirmed that this actually bloats the code? I would expect the linker to remove the duplication.

Reply:
> For Axel's question,
> The new x64 debug mrm.dll is 898KB in size.
> The old x64 debug mrm.dll is 840KB in size.
> 
> For Hui's question, redirecting to @ZenBird.

What are the release numbers?

Reply:
It seems the only thing the lib does is to load version and experiment level from a MUI dll. Does we even need that dll? can we just make it a header file with 2 macros instead? The build pipeline can update the macros with correct value.

Reply:
The header will have to be build in ProjectReunionInternal then. The Github pipeline does not have the version numbers. ProjectReunionInternal does not build the Github code. It only pulls the github compiled binaries and packages them. This is a cyclic dependency problem. MrtCore cannot depend on ProjectReunionInternal and vice-versa.

Reply:
I assume that's how telemetry works? The github code uses stub (placeholder). The internal pipeline replaces with the real thing. BTW, ProjectReunionInternal does build github code.

Reply:
Having a header only is possible. One solution is to publish a separate Reunion telemetry package that must be published by ProjectReunionInternal before Github code starts to build. This package needs to be updated for every single build. The Github pipline will need to pull this package before it starts building. This is unnecessarily complicated.

Reply:
The Release numbers as requested (by Axel):
The new x64 release mrm.dll is 315KB in size.
The old x64 release mrm.dll is 327KB in size.

Reply:
https://github.com/microsoft/WindowsAppSDK/pull/1268 switched impl to header-only